src/abstractarray.jl

```diff
@@ -286,3 +286,12 @@ end
         @inbounds return similar_type(a, promote_type(eltype(a), eltype(b)), Size($Snew))(tuple($(exprs...)))
     end
 end
+
+if VERSION >= v"1.6.0-DEV.1334"
+    # FIXME: This always assumes one-based linear indexing and that subtypes of StaticArray
+    # don't overload iterate
+    @inline function Base.rest(a::StaticArray, (_, i) = (nothing, 0))
+        newlen = StaticArrays.tuple_prod(Size(a)) - i
```

Comment:
```suggestion
    @inline function Base.rest(a::StaticArray{S}, (_, i) = (nothing, 0)) where {S}
        newlen = tuple_prod(S) - i
```
This should fix the failure on Julia master.